### ISSUE TITLE
PowerShell does not show output of `where python` command

### ISSUE THREAD

**Issue by Iftakharpy** [PowerShell/PowerShell#24662]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Run command `where python` on PowerShell and Command Prompt. There might be more commands that does not print output to the console.
![image](https://github.com/user-attachments/assets/5c61fa6a-3a21-4e2e-9c39-2386b12e090d)

### Expected behavior

```console
PowerShell should print the output of the command.
```

### Actual behavior

```console
PowerShell doesn't print the output of the command `where python`
```

### Error details

```console
N/A
```

### Environment data

```powershell
PS > $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.4.6
PSEdition                      Core
GitCommitId                    7.4.6
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by jborean93** [comment]
`where` in PowerShell is an alias for `Where-Object`

```
Get-Command where

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Alias           where -> Where-Object
```

You either have to add the `.exe` suffix if you want to use the `where.exe` executable as aliases take priority over them. You can also use `Get-Command` to find this information

```powershell
where.exe cmd
# C:\Windows\System32\cmd.exe

Get-Command cmd -CommandType Application

# CommandType     Name                                               Version    Source
# -----------     ----                                               -------    ------
# Application     cmd.exe                                            10.0.2610… C:\WINDOWS\system32\cmd.exe

(Get-Command cmd -CommandType Application).Path
# C:\WINDOWS\system32\cmd.exe
```

**Comment by ccrutchf** [comment]
Does it make sense to make `which` a default alias for `Get-Command`?

I have a lot of bash background, and my fingers often want to type `which`, so I have created an alias myself.  I think this would help the discoverability of `Get-Command` and fit with the existing precedent of `ls` being an alias over `Get-ChildItem`.

**Comment by 237dmitry** [comment]
>  I think this would help the discoverability of Get-Command

I think it's better to leave this concern to the users. For each new alias, there will definitely be a similar external utility. And this similarity will have to be remembered and taken into account. It's one thing when you personally assigned the alias, but quite another when it's out of the box. For example

```
$ gcm sort

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Alias           sort -> Sort-Object                                           

$ which sort
/usr/bin/sort
```

I use aliases only in interactive shell.

**Comment by mklement0** [comment]
Good points, @237dmitry, but it's worth pointing out that `sort` is only an alias for `Sort-Object` _on Windows_, given that a decision was made not to shadow a given platform's standard utilities (e.g. `/usr/bin/sort`on Unix-like platforms).

Allow me to recycle part of [this StackOverflow answer](https://stackoverflow.com/a/74591072/45375):

---

#### Why aliases named for a _different_ shell's commands - such as `cd` and `mkdir` - are *best avoided* in PowerShell:

In a problematic attempt to ease the migration pain for `cmd.exe` users (and in part also for users of POSIX-compatible shells), PowerShell decided to define built-in aliases and wrapper functions that are named for `cmd.exe`'s internal commands, whereas PowerShell's analogous internal commands, the so-called _cmdlets_, have very different names.

The latter isn't problematic per se - except that by their use you're missing out on the benefits of PowerShell's standard verb-noun naming convention (e.g., `Set-Location`), which also extends to how _aliases_ are formed, given that the [approved verbs](https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands) have official alias forms (e.g., `s` for `Set-`; therefore, `sl` is another, but *PowerShell-idiomatic* `Set-Location` alias).

What _is_ problematic, however, is that **PowerShell commands have [_very different syntax_](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Command_Syntax) from `cmd.exe`'s**.

If you use names such as `cd` and `mkdir`, you'll be tempted to think that, e.g. `cd` and `mkdir` function the same way as in `cmd.exe` - which is only true in the most basic of use cases, however.

**It's best to use the true PowerShell command names or - for brevity in *interactive* use - their _PowerShell_ aliases**, which are (reasonably) predictably formed, as discussed above (e.g., `sl` for `Set-Location` and `ni` for `New-Item`)

  * Note that use of _built-in_ aliases - as opposed to _user-defined_ ones - is safe to use in _script files_ too, though for clarity  you may choose full command names consistently.

---

In short:

* It's worth spending the time to learn how PowerShell-idiomatic alias names are formed, which by definition work on _all_ supported platforms.

* In the case at hand: use the built-in `gcm` alias of `Get-Command`.

* To generally discover a given cmdlet's _built-in_ aliases, use the following (using `Get-Command` as an example):

```powershell
& (Get-Process -Id $PID).Path -NoProfile { Get-Alias -Definition $args[0] }  -args Get-Command
```
